### PR TITLE
update LRO's intervalInMs from the Retry-After header

### PR DESCRIPTION
**Pull request by deyaaeldeen**
Fixes https://github.com/Azure/autorest.typescript/issues/1034.

The algorithm is simple, if the service's response has `Retry-After` header, use that instead of the customer-provided/default polling interval.

TODO: update core-lro if necessary to make sure it always use the updated intervalInMs.

### CODE REVIEW

**Review by sarangan12 — APPROVED** [review]

File: src/lro/stateMachine.ts [hunk 1]
```diff
@@ -133,12 +139,36 @@ export function createGetLROState<TResult>(
...
+  return async (
+    path: string,
+    pollerConfig: PollerConfig
+  ): Promise<LROState<TResult>> => {
     await poll(path);
+    if (retryAfter !== undefined) {
+      const retryAfterInMs = parseInt(retryAfter);
+      pollerConfig.intervalInMs = isNaN(retryAfterInMs)
+        ? calculatePollingIntervalFromDate(
+            new Date(retryAfter),
```
Comment (joheredi) [inline src/lro/stateMachine.ts:151]: Just double checking if you are passing `retryAfter` instead of `retryAfterInMs` on purpose.

Comment (joheredi) [reply src/lro/stateMachine.ts:151]: If we don't get neither pollingInterval, nor Retry-After. Are we setting a default value?

Comment (deyaaeldeen) [reply src/lro/stateMachine.ts:151]: For your first question, yes this is intended because `retryAfterInMs` is unusable since it is `NaN` in this branch.

**Review by joheredi — APPROVED** [review]
Thanks!

**Pull request merged**